packages/accordions/src/elements/stepper/components/Step.tsx

```diff
@@ -5,19 +5,34 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import React, { forwardRef, useRef, LiHTMLAttributes } from 'react';
+import React, { forwardRef, LiHTMLAttributes, useEffect, useState } from 'react';
 import { StyledStep, StyledLine } from '../../../styled';
 import { StepContext, useStepperContext } from '../../../utils';
 
+const useStep = () => {
```

Comment:
Even though this abstraction is helpful; if we aren't using this hook multiple places it might make more sense to have the logic live within `<Step />` directly. This also would require returning the remaining `context` values as they aren't involved in the step calculation.

Reply:
Happy to change this

Reply:
Let's bring this logic back into the component. Overall the logic looks good to me.